Refactoring and YARD documenting

Genki-S opened the pull request:
I determined to make code a little easier to understand, so I did some refactoring and wrote documentation with YARD.

@robin850 could you review the code please?

While doing this, I realized I made a bug, that is `TTNT::TestToCodeMapping#read_mapping` should read test-to-code mapping from the git tree of base object (i.e. mapping file saved before). Currently, it's reading test-to-code mapping from mapping file in current working tree. I will address this bug in different pull request. (issue: #16)

robin850 commented on lib/ttnt.rb:
```diff
@@ -1,5 +1,8 @@
 
+# Test This, Not That!
+#
+# See {file:README.md} for more detail.
```
s/detail/details

robin850 commented on lib/ttnt/test_selector.rb:
```diff
@@ -3,21 +3,27 @@
     end
 
+    # Select tests using differences in base_sha...target_sha and the latest
+    # TestToCodeMapping committed to base_sha
```
Missing dot here.

robin850 commented on lib/ttnt/test_to_code_mapping.rb:
```diff
@@ -2,23 +2,33 @@
-#   mapping: { test_file => spectra }
-
 module TTNT
+  # Mapping from test file to executed code (i.e. coverage without execution count)
```
Missing dot here as well.

robin850 commented on lib/ttnt/test_to_code_mapping.rb:
```diff
@@ -2,23 +2,33 @@
 
+    # Append the new mapping to test-to-code mapping file.
+    #
+    # @param test [String] test file to which the coverage data is produced
```
s/to/for

robin850 commented on lib/ttnt/test_to_code_mapping.rb:
```diff
@@ -41,6 +55,10 @@ def get_tests(file:, lineno:)
       tests
     end
 
+    # Save sha as the commit object anchoring has been run on
```
"Save commit's sha anchoring has been run on." (with a dot at the end :p)

Genki-S replied:
I'm surprised how broken this sentence was... Thanks!

robin850 commented on lib/ttnt/test_to_code_mapping.rb:
```diff
@@ -41,6 +55,10 @@ def get_tests(file:, lineno:)
 
+    # Save sha as the commit object anchoring has been run on
+    #
+    # @param sha [String] sha of the commit anchoring has been run on
```
Ditto above.

robin850 commented on lib/ttnt/version.rb:
```diff
@@ -1,3 +1,4 @@
 module TTNT
+  # TTNT version
```
This seems pretty useless, don't you think so ? :scissors:

Genki-S replied:
True. Sorry I added this in order to make documentation coverage (which you can produce with `yard stats`) 100%, but you are right it's useless :sweat_smile:  I removed it here: https://github.com/Genki-S/ttnt/commit/4ff64326dd2fdb830854f8d10dbded36520002c4

robin850 commented:
This looks pretty good, great work! :+1: I've left a some comments for a few nit-picky things. Here are a couple things too:

* Do you think that we need to document private methods ? These are pretty internal to the class so people can't use them inside their code. We can make some of them public later if the need for them grow.
* What do you think about renaming `#normalize_path` to `#normalize_paths` ? Pretty nit-picky but there are most-likely several paths inside a spectra.
* As for the Rake task, I think that we should make all methods private apart from `#initialize`. They are pretty internal and people won't need them in most cases.

Also please in the future, separate documentation pull requests from refactoring ones. There's no problem but this is more logical. :-)

Genki-S commented:
> I've left a some comments for a few nit-picky things

Thank you for catching my errors! I've fixed those in this commit: https://github.com/Genki-S/ttnt/commit/7265c25d5e553f1a7bd41f25934290cc5eca22df

> Do you think that we need to document private methods ?

For now, yes. To be honest my mind is so messy that I want to make things as clear as possible. Documenting all methods help me be conscious about what I'm doing. I might remove the burden of documenting private methods once this gem is stable enough to understand the logic only with method names, but I can manage it for now and it's helpful. So let me allow writing docs for a while :pray: 

> What do you think about renaming #normalize_path to #normalize_paths ? Pretty nit-picky but there are most-likely several paths inside a spectra.

True. Thank you for mentioning! Fixed here: https://github.com/Genki-S/ttnt/commit/a828cfa2b69d886896e137c8fa483a79d9bae4b5

> As for the Rake task, I think that we should make all methods private apart from #initialize

You are right, thanks! Fixed here: https://github.com/Genki-S/ttnt/commit/c8d13b7cb16e3940adca0592abdc33c0f8061712

> Also please in the future, separate documentation pull requests from refactoring ones

Got it! I'm sorry for the inconvenience :disappointed: I'll take care in the future! :+1: 

---

And I'm sorry Hound CI is hounding so loudly. I'll fix code style after I merge this PR :sweat:

robin850 commented:
> To be honest my mind is so messy that I want to make things as clear as possible.

Okay, seems fine to keep them then! :+1:

Genki-S commented:
Thank you for merging! :smile:
